src/test/java/book_store/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import java.util.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.authentication;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@WithMockUser(username = "user", roles = {"USER"})
+class ShoppingCartControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private ShoppingCartService shoppingCartService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @DisplayName("Get shopping cart")
+    void getShoppingCart() throws Exception {
+        ShoppingCartDto mockShoppingCartDto = createMockShoppingCartDto();
+        User user = new User();
+        Authentication authentication = new UsernamePasswordAuthenticationToken(user, null,
+                AuthorityUtils.createAuthorityList("ROLE_USER"));
+        when(shoppingCartService.findUserShoppingCart(eq(user), any(Pageable.class))).thenReturn(mockShoppingCartDto);
+        mockMvc.perform(get("/shoppingCarts").with(authentication(authentication)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(1))
+                .andExpect(jsonPath("$.cartItems.length()").value(2));
+    }
+
+    @Test
+    @DisplayName("Add book to shopping cart")
+    void addBookToShoppingCart() throws Exception {
+        AddToCartRequest request = new AddToCartRequest();
+        request.setBookId(123L);
+        request.setQuantity(1);
+        ShoppingCartDto mockShoppingCartDto = createMockShoppingCartDto();
+        User user = new User();
+        Authentication authentication = new UsernamePasswordAuthenticationToken(user, null,
+                AuthorityUtils.createAuthorityList("ROLE_USER"));
+        when(shoppingCartService.addBookToCart(any(AddToCartRequest.class), eq(user))).thenReturn(mockShoppingCartDto);
+        mockMvc.perform(post("/shoppingCarts")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(request))
+                        .with(authentication(authentication)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(1))
+                .andExpect(jsonPath("$.cartItems.length()").value(2));
+    }
+
+    @Test
+    @DisplayName("Update the quantity of goods in the cart")
+    void updateCartItemQuantity() throws Exception {
```

Comment:
test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`